libs/category/state/src/reducers/category/pure/build-metadata-from-request.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  DaffCategoryPageMetadata,
+  DaffCategoryRequestReplacement,
+} from '@daffodil/category';
+
+type MetadataRequestIntersection = keyof DaffCategoryPageMetadata & keyof DaffCategoryRequestReplacement;
+
+const allowedMetadataKeys: MetadataRequestIntersection[] = [
+  'id',
+  'applied_sort_option',
+  'applied_sort_direction',
+  'current_page',
+  'page_size',
+];
+
+export const buildMetadataFromRequest = (request: DaffCategoryRequestReplacement): Partial<DaffCategoryPageMetadata> =>
```

Comment:
Needs tests. Could you make sure to add a test like "should only return keys included in the request"?

Reply:
Its tested in the [reducer test](https://github.com/graycoreio/daffodil/blob/73ad326f65eb2bb4e5975c8a7f13ed9114ea885b/libs/category/state/src/reducers/category/category.reducer.spec.ts#L930). Since its not in the public API I don't think it needs a standalone test.